src/ticks.jl

```diff
@@ -183,176 +213,135 @@ function optimize_ticks_typed(
+                        c = if len > 1
+                            effective_span = (len - 1) * tickspan
+                            1.5xspan / effective_span
+                        else
+                            F(0)
+                        end
 
-                    score =
-                        granularity_weight * g +
-                        simplicity_weight * s +
-                        coverage_weight * c +
-                        niceness_weight * qscore
+                        score =
+                            granularity_weight * g +
+                            simplicity_weight * s +
+                            coverage_weight * c +
+                            niceness_weight * qscore
 
-                    # strict limits on coverage
-                    if strict_span && span > xspan
-                        score -= 10000
-                    end
-                    if span >= 2xspan
-                        score -= 1000
-                    end
+                        # strict limits on coverage
+                        if strict_span && span > xspan
+                            score -= 10000
+                        end
+                        if span >= 2xspan
+                            score -= 1000
+                        end
 
-                    if score > high_score && (k_min <= len_S <= k_max)
-                        if strict_span
-                            # make S a copy because it is a view and
-                            # could otherwise be mutated in the next runs
-                            S = collect(S)
+                        if score > high_score && (k_min <= len <= k_max)
+                            viewmin_best, viewmax_best = viewmin, viewmax
+                            high_score, len_S_best = score, len
+                            copyto!(S_best, view(S, 1:len))
                         end
-                        S_best, viewmin_best, viewmax_best = S, viewmin, viewmax
-                        high_score = score
+                        r += 1
                     end
-                    r += 1
                 end
             end
-        end
-        z -= 1
-    end
-
-    if isinf(high_score)
-        if strict_span
-            @warn "No strict ticks found"
-            return optimize_ticks_typed(
-                x_min,
-                x_max,
-                extend_ticks,
-                Q,
-                k_min,
-                k_max,
-                k_ideal,
-                granularity_weight,
-                simplicity_weight,
-                coverage_weight,
-                niceness_weight,
-                false,
-                span_buffer,
-                scale,
-            )
-        else
-            return fallback_ticks(x_min, x_max, k_min, k_max)
+            z -= 1
         end
     end
 
-    return S_best, viewmin_best, viewmax_best
+    return high_score, S_best[1:len_S_best], viewmin_best, viewmax_best
```

Comment:
using `resize!(S_best, len_S_best)` and returning `S_best` might be faster here, in theory this saves allocation of one vector.

Reply:
Done. Very nice, thanks !

Reply:
The previous comment `# 816` was wrong (certainly from another julia version). Using `1.6` it went down from `944` to `848`.

Reply:
Merge ?

Reply:
Did you run the Plots test-suite with this? If so, I am fine with merging

Reply:
I've ran it, all ref images and test do pass locally.